packages/react-redux/src/slices/cartoSlice.js

```diff
@@ -78,9 +78,16 @@ export const createCartoSlice = (initialState) => {
       updateLayer: (state, action) => {
         const layer = state.layers[action.payload.id];
         if (layer)
+          // TODO: Study if we should use a deepmerge fn
           state.layers[action.payload.id] = {
             ...layer,
-            ...action.payload.layerAttributes
+            ...action.payload.layerAttributes,
+            ...(layer.legend && {
```

Comment:
i think state, we're unable to _add_ legent unless one already exists, not sure if that was intended
looks more like it should be:

```
(layer.legend || action.payload.layerAttributes.legend) &&  { legend: { ... }}
```

this version on the other hand prevents from removing legend at all

Reply:
If legend doesn't exist, it's added when `action.payload.attributes` is merged.